extensions/podman/src/extension.ts

```diff
@@ -116,10 +116,33 @@ async function updateMachines(provider: extensionApi.Provider): Promise<void> {
     connectionsToCreate.map(async machineName => {
       // podman.sock link
       let socketPath;
-      if (isMac) {
-        socketPath = calcMacosSocketPath(machineName);
-      } else if (isWindows) {
-        socketPath = calcWinPipeName(machineName);
+      try {
+        if (isWindows) {
+          socketPath = await execPromise(getPodmanCli(), [
+            'machine',
+            'inspect',
+            '--format',
+            '{{.ConnectionInfo.PodmanPipe.Path}}',
+            machineName,
+          ]);
+        } else {
+          socketPath = await execPromise(getPodmanCli(), [
+            'machine',
+            'inspect',
+            '--format',
+            '{{.ConnectionInfo.PodmanSocket.Path}}',
+            machineName,
+          ]);
+        }
+      } catch (error) {
+        socketPath = undefined;
```

Comment:
In case of error socketPath will stay undefined
probably just log it to the debug console

Reply:
Applied this diff to the same commit
```diff
@@ -133,13 +133,13 @@ async function updateMachines(provider: extensionApi.Provider): Promise<void> {
             '{{.ConnectionInfo.PodmanSocket.Path}}',
             machineName,
           ]);
         }
       } catch (error) {
-        socketPath = undefined;
+        console.debug('Podman extension:', 'Failed to read socketPath from machine inspect')
       }
-      if (socketPath === undefined) {
+      if (!socketPath) {
         if (isMac) {
           socketPath = calcMacosSocketPath(machineName);
         } else if (isWindows) {
           socketPath = calcWinPipeName(machineName);
         }
```